azure-pipelines.yml

```diff
@@ -186,7 +191,31 @@ jobs:
           AZCOPY_E2E_STD_MANAGED_DISK_CONFIG: $(AZCOPY_E2E_STD_MANAGED_DISK_CONFIG)
           CPK_ENCRYPTION_KEY: $(CPK_ENCRYPTION_KEY)
           CPK_ENCRYPTION_KEY_SHA256: $(CPK_ENCRYPTION_KEY_SHA256)
-        displayName: 'E2E Test MacOs'
+        displayName: 'E2E Test MacOs AMD64'
+        condition: eq(variables.type, 'mac-os')
+
+      # Running E2E Tests on Mac
+      - script: |
+          set -e
+          go build -o azcopy_darwin_arm64
+          echo 'starting E2E tests on mac-os'
+          export AZCOPY_E2E_EXECUTABLE_PATH=$(pwd)/azcopy_darwin_arm64
+          go test -timeout 60m -race -cover -v ./e2etest
```

Comment:
I assume the devops pipeline does not yet support Mac M1 os, Are we testing M1 build here?

Reply:
yes

Reply:
yea